client/src/image-uploader.tsx

```diff
@@ -2,29 +2,61 @@ import { useCallback, useContext } from "react";
 import { AppContext } from "./app-context";
 import { useDropzone } from "react-dropzone";
 import { Box, Grid, Typography, useTheme } from "@mui/joy";
+import { imageService } from "./services/imageService";
```

Comment:
we don't need to modify image-uploader too. This component must only get an image from the user and update the AppContext, which it already does pretty well.

Image upscaling as I already mentioned must be implemented in a filters panel